ludwig/benchmarking/utils.py

```diff
@@ -0,0 +1,125 @@
+
+    config: ludwig config.
+    """
+    config_dict = copy.deepcopy(config)
+    config = merge_with_defaults(config_dict)
+    return config
+
+
+def load_from_module(
+    dataset_module: ModuleType, output_feature: Dict[str, str], subsample_frac: float = 1
+) -> Tuple[pd.DataFrame, pd.DataFrame, pd.DataFrame, pd.DataFrame]:
+    """Take in Ludwig Python module of the dataset, load it, and split it if it isn't already (by adding "split"
+    column).
+
+    dataset_module: ludwig dataset module (e.g. ludwig.datasets.sst2, ludwig.datasets.ames_housing, etc.)
+    subsample_frac: percentage of the total datasets to load.
+    """
+    dataset = dataset_module.load(split=False)
+    if subsample_frac < 1:
+        dataset = dataset.sample(frac=subsample_frac, replace=False, random_state=default_random_seed)
+
+    if output_feature["type"] == CATEGORY:
+        dataset = get_repeatable_train_val_test_split(
+            dataset,
+            stratify_colname=output_feature["name"],
+            random_seed=default_random_seed,
+            frac_train=0.75,
+            frac_val=0.1,
+            frac_test=0.15,
+        )
+    else:
+        dataset = get_repeatable_train_val_test_split(
+            dataset, random_seed=default_random_seed, frac_train=0.75, frac_val=0.1, frac_test=0.15
+        )
+
+    train_df = dataset[dataset["split"] == 0]
+    val_df = dataset[dataset["split"] == 1]
+    test_df = dataset[dataset["split"] == 2]
+    return train_df, val_df, test_df, dataset
+
+
+def bytes_to_human_readable(num: Union[float, int], suffix: str = "B") -> str:
+    """Transform bytes into a human readable format."""
+    for unit in ["", "Ki", "Mi", "Gi", "Ti", "Pi", "Ei", "Zi"]:
+        if abs(num) < 1024.0:
+            return f"{num:3.1f}{unit}{suffix}"
+        num /= 1024.0
+    return f"{num:.1f}Yi{suffix}"
+
+
+def flatten_dict(d: Dict[str, Any], sep: str = ".") -> Dict[str, Any]:
+    [flat_dict] = pd.json_normalize(d, sep=sep).to_dict(orient="records")
+    return flat_dict
+
+
+def export_artifacts(
+    experiment: Dict[str, str], report_path: str, experiment_output_directory: str, export_base_path: str
+) -> None:
+    """Save the experiment artifacts to the `bench_export_directory`.
+
+    experiment: experiment dict that contains dataset name,
+        experiment name, and config path.
+    report_path: path where the experiment metrics report is
+        saved.
+    experiment_output_directory: path where the model, data,
+        and logs of the experiment are saved.
+    export_base_path: path (directory) where artifacts are
+        exported. (e.g. s3://benchmarking.us-west-2.ludwig.com/bench/)
+    """
+    protocol, _ = fsspec.core.split_protocol(export_base_path)
+    fs, _ = get_fs_and_path(export_base_path)
+    try:
+        experiment_name = (
+            experiment["experiment_name"]
+            if experiment["experiment_name"]
+            else datetime.now().strftime("%Y.%m.%d.%H:%M:%S")
```

Comment:
What's the reason for not appending a timestamp to the experiment name always?

Reply:
One potential reason is that it would be easier to retrieve the experiment report (or filter by experiment name) without having the date there. 
Example: we want to benchmark a new version (e.g. `0.5.4`) of ludwig on a number of datasets. If we later want to compare all `0.5.3` vs. `0.5.4` benchmarks, it would be easier to filter by this name without the date. The dated names will be `0.5.4-2022.07.12.11:23:44`, `0.5.4-2022.07.12.11:31:19` and it might be more of a challenge to use these names. However, we can make the timestamp available in the exported report.

What do you think?